prow/cmd/crier/main.go

```diff
@@ -253,7 +245,7 @@ func main() {
 	}
 
 	if o.gerritWorkers > 0 {
-		gerritReporter, err := gerritreporter.NewReporter(cfg, o.cookiefilePath, o.gerritProjects, mgr.GetCache())
+		gerritReporter, err := gerritreporter.NewReporter(cfg, o.cookiefilePath, o.gerritProjects, mgr.GetClient())
```

Comment:
With the changes from this PR, gerrit reporter need to update previous report states, so it cannot just be a reader

Reply:
Had an offline chat with @cjwagner , understand more about the concern here: is the client from `GetClient()` smart enough to use cache instead of calling api server? Based on GoDoc it seems like that the default Client uses cache https://github.com/kubernetes-sigs/controller-runtime/blob/f236f0345ad2933912ebf34bfcf0f93620769654/pkg/cluster/cluster.go#L266, if that's the case here then we might be fine. A question not very clear is: do we need to initialize the cache like https://github.com/kubernetes/test-infra/blob/66c01a3f217e7aac1afef2974a6f9b7ae71d75e0/prow/plank/reconciler.go#L95?

@alvaroaleman , did I understand it correctly?

Reply:
Yeah, the client obtained through `GetClient` reads from the cache and writes onto the api by default. You don't need to initialize anything for that. It will automatically set up a cache for every object kind you request through it if it hasn't one yet.

The code you linked above sets up an index, which improves performance if you want to look up objects by something other than name or labels as you don't have to do a linear search over all objects.

Reply:
awesome! So this should just work!